javascript-client/src/client.ts

```diff
@@ -68,11 +71,36 @@ export default class S3FileFieldClient {
       apiConfig = {},
     }: S3FileFieldClientOptions,
   ) {
+    // Create axios instances for API and direct S3 requests
     this.api = axios.create({
       ...apiConfig,
       // Add a trailing slash
       baseURL: baseUrl.replace(/\/?$/, '/'),
     });
+    this.s3 = axios.create();
+
+    // Attach axios-retry to both axios instances
+    const raxConfig = {
+      // Retry 3 times on requests that return a response (500, etc) before giving up.
+      retry: 3,
+      // Retry twice on errors that don't return a response (ENOTFOUND, ETIMEDOUT, etc).
+      noResponseRetries: 2,
+      httpMethodsToRetry: ['GET', 'HEAD', 'OPTIONS', 'DELETE', 'PUT', 'POST', 'PATCH'],
+      statusCodesToRetry: [[429, 429], [500, 599]],
+      onRetryAttempt: (err: AxiosError) => {
+
```

Comment:
Unfortunately, this library just provides a global handler for the axios instance. I'm not sure if there's an easy way to notify the per-upload progress callback that **their** upload is currently retrying.

Is it an important feature that the progress callback actually communicate a retrying state, particularly given the fact that retries here will not continue indefinitely?